spec/checker_spec.lua

```diff
@@ -60,32 +60,6 @@ describe("Pallene type checker", function()
         assert(prog_ast, errs)
```

Comment:
It is missing tests for the value type.

Reply:
I added some tests to guarantee code coverage for the implicit cast insertion. But I am not super duper confident that I added all the tests I could for value types in general. We might want to pay some attention to this before merging.